llvm/include/llvm/DebugInfo/DWARF/DWARFTypePrinter.h

```diff
@@ -9,19 +9,37 @@
 #ifndef LLVM_DEBUGINFO_DWARF_DWARFTYPEPRINTER_H
 #define LLVM_DEBUGINFO_DWARF_DWARFTYPEPRINTER_H
 
+#include "llvm/ADT/SmallString.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/BinaryFormat/Dwarf.h"
-#include "llvm/DebugInfo/DWARF/DWARFDie.h"
+#include "llvm/DebugInfo/DWARF/DWARFUnit.h"
```

Comment:
This this actually need/depend on llvm::DWARFUnit (in a non-generic way)? Because, if it does, then it probably won't work with lldb. (and if it does not, then it shouldn't need to include it)

Reply:
Updated, it's not needed. DWARFFormValue.h is actually needed here.

Reply:
Okay, but that's also going to be a problem, since lldb has its own DWARFFormValue as well. It sounds to me like you need an indirect way to refer to the DWARFFormValue type. The way that's usually done is through typedefs.

We could have the DWARFDie class provide a typedef for the form value version that is compatible with it. Or, if that's awkward for some reason, we could put the both DWARFDie and DWARFFormValue typedefs into the DWARFUnit class, and then parametrize the algorithm on the unit type.

Reply:
Removed dependency on `llvm:: DWARFFormValue`. I didn't noticed it when splitting it from the original PR.